src/k8s-engine/kim.js

```diff
@@ -0,0 +1,173 @@
+'use strict';
+
+// This file should probably be an instance of ImageManager < EventEmitter.
+// We'll get there eventually
+
+const { EventEmitter } = require('events');
+const process = require('process');
+const { spawn } = require('child_process');
+const os = require('os');
+const fs = require('fs');
+const path = require('path');
+const util = require('util');
+const paths = require('xdg-app-paths')({ name: 'rancher-desktop' });
+const resources = require('../resources');
+const K8s = require('./k8s');
+
+const REFRESH_INTERVAL = 5 * 1000;
+
+async function runCommand(args) {
+  const child = spawn(resources.executable('kim'), args);
+  const result = { stdout: '', stderr: '' };
+
+  return await new Promise((resolve, reject) => {
+    child.stdout.on('data', (data) => {
+      result.stdout += data.toString();
+    });
+    child.stderr.on('data', (data) => {
+      result.stderr += data.toString();
+    });
+    child.on('exit', (code, sig) => {
+      if (code === 0) {
+        resolve(result);
+      } else if (sig !== undefined) {
+        reject({ ...result, signal: sig });
+      } else {
+        reject(result);
+      }
+    });
+  });
+}
+
+class Kim extends EventEmitter {
+  constructor() {
+    super();
+    this.notifiedMissingKim = false;
+    this.showedStderr = false;
+  }
+
+  start() {
+    fs.access(resources.executable('kim'),
+      fs.constants.R_OK | fs.constants.X_OK,
+      (err) => {
+        if (err) {
+          if (!this.notifiedMissingKim) {
+            const dirname = path.dirname(resources.executable('kim'));
+
+            console.log(`\nkim executable not found in ${ dirname }`);
+            this.notifiedMissingKim = true;
+          }
+        } else {
+          this.refreshImages();
+        }
+      });
+  }
+
+  stop() {
+    if (this.refreshInterval) {
+      clearInterval(this.refreshInterval);
+    }
+  }
+
+  async buildImage(dirPart, filePart, taggedImageName) {
+    try {
+      const args = ['build'];
+
+      if (filePart !== 'Dockerfile') {
+        args.push('--file');
+        args.push(filePart);
+      }
+      args.push('--tag');
+      args.push(taggedImageName);
+      args.push(dirPart);
+
+      return await runCommand(args);
+    } catch (err) {
+      console.log(`Error building image ${ taggedImageName }:`);
+      console.log(err.stderr);
+
+      return err;
+    }
+  }
+
+  async deleteImage(imageID) {
+    try {
+      return await runCommand(['rmi', imageID]);
+    } catch (err) {
+      console.log(`Error deleting image ${ imageID }:`);
+      console.log(err.stderr);
+
+      return err;
+    }
+  }
+
+  async pullImage(taggedImageName) {
+    try {
+      return await runCommand(['pull', taggedImageName]);
+    } catch (err) {
+      console.log(`Error pulling image ${ taggedImageName }:`);
```

Comment:
Did you mean to log the error here? (Also the next function.)

Reply:
Changed with pushing try/catch to `background.ts`